Content.Server/GameObjects/Components/Nutrition/SmokingComponent.cs

```diff
@@ -0,0 +1,110 @@
+﻿#nullable enable
+using System.Threading.Tasks;
+using Content.Server.GameObjects.Components.Items.Clothing;
+using Content.Shared.GameObjects.Components;
+using Content.Shared.Interfaces.GameObjects.Components;
+using Robust.Server.GameObjects;
+using Robust.Shared.GameObjects;
+using Robust.Shared.GameObjects.ComponentDependencies;
+using Robust.Shared.GameObjects.Components.Timers;
+using Robust.Shared.Serialization;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components.Nutrition
+{
+    /// <summary>
+    /// This item acts as a representation for smokable consumables.
+    ///
+    /// To smoke a cigar, you need:
+    /// <list type="bullet">
+    /// <item><description> a hot item (implements IHotItem interface)</description></item>
+    /// <item><description> that's a alight.</description></item>
+    /// <item><description>  for the target cigar be Unlit. Lit cigars are already lit and butt's don't have any "fuel" left.</description></item>
+    ///</list>
+    /// TODO: Add reagents that interact when smoking
+    /// TODO: Allow suicide via excessive Smoking
+    /// </summary>
+    [RegisterComponent]
+    public class SmokingComponent : Component, IInteractUsing, IHotItem
+    {
+        public override string Name => "Smoking";
+
+        private SharedBurningStates _currentState = SharedBurningStates.Unlit;
+
+        [ComponentDependency] private readonly ClothingComponent? _clothingComponent = default!;
+        [ComponentDependency] private AppearanceComponent? _appearanceComponent;
+
+        /// <summary>
+        /// Duration represents how long will this item last.
+        /// Generally it ticks down whether it's time-based
+        /// or consumption-based.
+        /// </summary>
+        [ViewVariables(VVAccess.ReadOnly)] private int _duration;
```

Comment:
Readonly is default, can just use the empty ctor
```suggestion
        [ViewVariables] private int _duration;
```